source/extensions/filters/common/rbac/matchers.cc

```diff
@@ -142,6 +145,26 @@ bool IPMatcher::matches(const Network::Connection& connection, const Envoy::Http
   case DownstreamLocal:
     ip = info.downstreamAddressProvider().localAddress();
     break;
+  case Upstream: {
+    if (!info.filterState().hasDataWithName(StreamInfo::KEY_DYNAMIC_PROXY_UPSTREAM_ADDR)) {
+      ENVOY_LOG_MISC(info, "Did not find dynamic forward proxy cookie. Do you have dynamic "
+                           "forward proxy in the filter chain?");
+      return false;
+    }
+
+    bool ipMatch = false;
+
+    const StreamInfo::AddressSetAccessor& address_set =
+        info.filterState().getDataReadOnly<StreamInfo::AddressSetAccessor>(
+            StreamInfo::KEY_DYNAMIC_PROXY_UPSTREAM_ADDR);
+    address_set.iterate([&, this](Network::Address::InstanceConstSharedPtr address) {
+      ipMatch = range_.isInRange(*address.get());
```

Comment:
If there were multiple entries (which is nice to support) will the final match depend on the order they're presented in?
I think per issue discussion you want to block if any of the addresses match, not simply if the last one does.  Can you fix, document in the API, and unit test?

Reply:
Ahh good catch. Will fix it.

Reply:
Thanks for the fix!  Mind also making it clear in the API that if any resolved addresses matched, that rbac will deny, rather than Envoy using one of the non-matching resolutions?